### ISSUE TITLE
What is InputPeer

### ISSUE THREAD

**Issue by kudla** [gram-js/gramjs#110]
Please in doc examples ([e.g.](https://painor.gitbook.io/gramjs/working-with-messages/messages.search)) there're references like
```
peer: new Api.InputPeer({...}),
```
But I can't find any docs on `InputPeer` it self. What is the way I should create it?

**Comment by rojserbest** [comment]
Why should you even create `InputPeer`?

**Comment by kudla** [comment]
Regarding creation this is what was mentioned in [docs](https://painor.gitbook.io/gramjs/working-with-messages/messages.search). 
Any way I'm just learning gramjs trying to search trough the channel chat with invocation of
```
new Api.messages.Search({
            peer:  ....,
            q: 'text',
            limit: 100
          })
```
and for peer I was trying different cases and got different errors
* exact chat id number - `Error: Could not find the input entity for [object Object]`
*  '@ChanelName' - `TypeError: Cannot read property 'getBytes' of undefined`
* channel instance from `Api.channels.GetChannels` invocation - `TypeError: Cannot read property 'getBytes' of undefined` as well

so the errors are not such clear to me. I'm not sure whether I'm using wrong notation or something wrong with my channel?

**Comment by painor** [comment]
The 

> Regarding creation this is what was mentioned in [docs](https://painor.gitbook.io/gramjs/working-with-messages/messages.search).
> Any way I'm just learning gramjs trying to search trough the channel chat with invocation of
> 
> ```
> new Api.messages.Search({
>             peer:  ....,
>             q: 'text',
>             limit: 100
>           })
> ```
> 
> and for peer I was trying different cases and got different errors
> 
> * exact chat id number - `Error: Could not find the input entity for [object Object]`
> * '@ChanelName' - `TypeError: Cannot read property 'getBytes' of undefined`
> * channel instance from `Api.channels.GetChannels` invocation - `TypeError: Cannot read property 'getBytes' of undefined` as well
> 
> so the errors are not such clear to me. I'm not sure whether I'm using wrong notation or something wrong with my channel?

The errors aren't really related to the peer part. The cannot read property error means you are missing a required arg of that request and the first error means that the library never saw that ID before.

The best way to pass a peer is using the third point you mentioned (GetChannels). 

I agree that the errors aren't clear enough. I'll try to figure out a way to make them more meaningful. 

so in short you need to pass all the required args. (or use https://gram.js.org/beta/classes/client_telegramclient.telegramclient.html#getmessages )

**Comment by kudla** [comment]
Cool. Thx for a clarification. I'll keep on diving
